models/db/install/dbinstall.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package install
+
+import (
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/xorm"
+)
+
+func getXORMEngine() *xorm.Engine {
+	return db.DefaultContext.(*db.Context).Engine().(*xorm.Engine)
+}
+
+// CheckDatabaseConnection checks the database connection
+func CheckDatabaseConnection() error {
+	e := db.GetEngine(db.DefaultContext)
+	_, err := e.Exec("SELECT 1")
```

Comment:
Why not use Ping?

Reply:
db.Engine doesn't have `Ping` ....

That's a little annoying that `db.Engine` lacks `IsTableExists` / `Ping` .... Maybe I will have to use xorm.Engine here too.

Reply:
And I do not like what sqlite ping does, I think `SELECT 1` is safer.

![image](https://user-images.githubusercontent.com/2114189/143183369-adde45e7-1ec6-472f-96cf-56f8164059d7.png)